dai_cadcad/flip.py

```diff
@@ -0,0 +1,119 @@
+""" Flip Module
+    Class-based representation of the Flip smart contract
+    (contains only what is necessary for the simulation)
+"""
+
+from dai_cadcad.util import require
+
+
+class Flipper:
+    ADDRESS = ""
+
+    beg = None
+    bids = {}
+    cat = None
+    ilk_id = None
+    kicks = None
+    ttl = None
+    tau = None
+    vat = None
+
+    def __init__(self, beg, bids, cat, ilk_id, kicks, ttl, tau, vat):
+        """"""
+
+        self.ADDRESS = f"flipper_{ilk_id}"
+
+        self.vat = vat
+        self.cat = cat
+        self.beg = beg
+        self.ttl = ttl
+        self.tau = tau
+        self.kicks = kicks
+        self.bids = bids
+        self.ilk_id = ilk_id
+
+    def kick(
+        self, usr, gal, tab, lot, bid, end
+    ):  # TODO: should these be instance attribute?
+        """Kicks off a new Flip auction."""
+
+        self.kicks += 1
+        bid_id = self.kicks
+
+        self.bids[bid_id] = {
```

Comment:
hmm I think we should make a `Bid` class instead of just using a dict, there's a struct defined for it in the [source code](https://github.com/makerdao/dss/blob/master/src/flip.sol#L58) and we've already adopted the convention of creating classes for structs that are defined in the source code (e.g. Ilk classes)

Reply:
and we should also save the bid's id as a class member on itself for convenience